ddl/ddl_api.go

```diff
@@ -2305,6 +2339,68 @@ func (d *ddl) AddColumn(ctx sessionctx.Context, ti ast.Ident, spec *ast.AlterTab
 	return errors.Trace(err)
 }
 
+// AddColumns will add multi new columns to the table.
+func (d *ddl) AddColumns(ctx sessionctx.Context, ti ast.Ident, specs []*ast.AlterTableSpec) error {
+	schema, t, err := d.getSchemaAndTableByIdent(ctx, ti)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	// Check all the columns at once.
+	newColumnsCount := 0
+	addingColumnNames := make(map[string]bool)
+	for _, spec := range specs {
+		for _, specNewColumn := range spec.NewColumns {
+			newColumnsCount++
+			if !addingColumnNames[specNewColumn.Name.Name.L] {
+				addingColumnNames[specNewColumn.Name.Name.L] = true
+			} else {
+				return errors.Trace(infoschema.ErrColumnExists.GenWithStackByArgs(specNewColumn.Name.Name.O))
```

Comment:
This seems unnecessary because `checkAddColumnTooManyColumns` will handle the duplication. On the other hand, consider:
```
create table t (a int);
alter table t add column b int, add column if not exists b int;
```
Our expected result is `Query OK`, which is consistent with MariaDB.

Reply:
Hi, there are two kind of duplication:
create table t(a int);
1. alter table t add column(a1 int, a1 long); // duplication each other
2. alter table t add column a int; // duplication with columns already have

Reply:
I will fix `if not exists`

Reply:
Hi, `if not exists` for add columns and `if exists ` for drop columns are supported now.